### PR TITLE
Battle map generation

### PR DESCRIPTION
**Pull request by PossibleLlama**
Added class for storing information related to the map, as well as one for generating the map.

### CODE REVIEW

**Comment by PossibleLlama** [comment]
Quick change to the BMG file so that the Map is returned.

File: project_kaccag/BattleMap/BattleMapGeneration.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+namespace project_kaccag.BattleMap
+{
+    public class BattleMapGeneration
+    {
+        private static Random rand;
+        public BattleMap map;
+        
+        private int maxNumberOfFactions;
+        // Should be an odd integer.
+        private int minSideSizeForFaction;
```
Comment (andreazhang) [inline project_kaccag/BattleMap/BattleMapGeneration.cs]: I don't understand what this is for. Is is faction supposed to be a square with odd side?

File: project_kaccag/BattleMap/BattleMapGeneration.cs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace project_kaccag.BattleMap
+{
+    public class BattleMapGeneration
+    {
+        private static Random rand;
```
Comment (andreazhang) [inline project_kaccag/BattleMap/BattleMapGeneration.cs]: Rather then creating a new instance in each constructor, you could create the new instance here and use it wherever you need.

File: project_kaccag/BattleMap/BattleMapGeneration.cs (hunk 1 above)
Comment (PossibleLlama) [reply project_kaccag/BattleMap/BattleMapGeneration.cs]: a a a
a b a
a a a

It makes more sense for a square to have a centre than for the centre to be between two tiles.

File: project_kaccag/BattleMap/BattleMapGeneration.cs (hunk 2 above)
Comment (PossibleLlama) [reply project_kaccag/BattleMap/BattleMapGeneration.cs]: I didn't know that that would work. I haven't been able to get anything to run so can't test any of this.

Comment (PossibleLlama) [reply project_kaccag/BattleMap/BattleMapGeneration.cs]: Having reread what you put here, I am once again confused.

File: project_kaccag/BattleMap/BattleMapGeneration.cs (hunk 1 above)
Comment (andreazhang) [reply project_kaccag/BattleMap/BattleMapGeneration.cs]: So each faction 'b' is always surrounded by tiles 'a'?
But a faction of size 2x2 can be surrounded by a 4x4 square of tiles.
I still don't get it...

File: project_kaccag/BattleMap/BattleMapGeneration.cs (hunk 2 above)
Comment (andreazhang) [reply project_kaccag/BattleMap/BattleMapGeneration.cs]: private static Random rand = new Random();
Delete the three new Random() calls in each constructor.

File: project_kaccag/BattleMap/BattleMapGeneration.cs (hunk 1 above)
Comment (PossibleLlama) [reply project_kaccag/BattleMap/BattleMapGeneration.cs]: a faction will be based around a single point. If you have an even number which side do you put the extra column and row on?

**Comment by PossibleLlama** [comment]
Done what I remember of the changes suggested at the review meeting. A quick check to make sure I haven't messed the changes and people are happy with this version.

**Pull request merged**